packages/cloudflare/src/entrypoints/middleware.ts

```diff
@@ -0,0 +1,15 @@
+import { When, whenAmI } from '@it-astro:when';
+import type { MiddlewareHandler } from 'astro';
+
+const middlewares: Record<any, MiddlewareHandler> = {
+	[When.Prerender]: (ctx, next) => {
+		if (ctx.locals.runtime === undefined) {
+			ctx.locals.runtime = {
+				env: process.env,
+			};
+		}
+		return next();
+	},
+};
+
+export const onRequest = middlewares[whenAmI];
```

Comment:
I _think_ Astro will fail to load an undefined injected middleware.

```suggestion
export const onRequest = middlewares[whenAmI] ?? ((_, next) => next());
```

I haven't tried it, if it works just ignore this.